### PR TITLE
release references except the result V8Object

### PR DESCRIPTION
**Pull request by ben-zhang-at-salesforce**
toV8(serializer: Serialization...) converts kotlin object into V8Object. the root object could have children objects. We should close references for newly created children objects and only keep the reference for root V8Object.

### CODE REVIEW

File: k2v8/src/main/java/com/salesforce/k2v8/K2V8.kt [hunk 1]
```diff
@@ -16,14 +17,16 @@ import kotlinx.serialization.modules.SerialModule
...
+class K2V8(val configuration: Configuration, override val context: SerialModule = DefaultModule) :
+    SerialFormat {
 
     /**
      * Serializes a [T] value to a [V8Object] using the [serializer] provided.
      */
-    fun <T> toV8(serializer: SerializationStrategy<T>, value: T): V8Object {
-        return convertToV8Object(value, serializer)
-    }
+    fun <T> toV8(serializer: SerializationStrategy<T>, value: T) =
```
Comment (ben-zhang-at-salesforce) [inline k2v8/src/main/java/com/salesforce/k2v8/K2V8.kt:26]: this is the core change, only keep the root v8 object and make sure the children objects reference released.

Comment (ben-zhang-at-salesforce) [reply k2v8/src/main/java/com/salesforce/k2v8/K2V8.kt:26]: As shown in the below test. 59 references should be released. image

**Review by mattkranzler5 — APPROVED** [review]
Looks good!

File: k2v8/src/main/java/com/salesforce/k2v8/V8ObjectDecoder.kt [hunk 2]
```diff
@@ -28,8 +29,13 @@ internal fun <T> K2V8.convertFromV8Object(
     value: V8Object,
     deserializer: DeserializationStrategy<T>
 ): T {
-    val decoder = V8ObjectDecoder(this, value)
```
Comment (ben-zhang-at-salesforce) [inline k2v8/src/main/java/com/salesforce/k2v8/V8ObjectDecoder.kt:31]: dig deeper and found a leak in the fromV8Object

image

**Review by threeve — APPROVED** [review]

**Pull request merged**